src/twisted/conch/test/test_transport.py

```diff
@@ -456,11 +459,25 @@ class BaseSSHTransportTests(BaseSSHTransportBaseCase, TransportTestCase):
     def test_sendVersion(self):
         """
         Test that the first thing sent over the connection is the version
-        string.
+        string.  The 'softwareversion' part must consist of printable
+        US-ASCII characters, with the exception of whitespace characters and
+        the minus sign.
+
+        RFC 4253, section 4.2.
         """
         # the other setup was done in the setup method
-        self.assertEqual(self.transport.value().split(b'\r\n', 1)[0],
-                         b"SSH-2.0-Twisted")
+        version = self.transport.value().split(b'\r\n', 1)[0]
+        self.assertEqual(version,
+                         b"SSH-2.0-Twisted_" + twisted_version.encode('ascii'))
+        softwareVersion = version.decode('ascii')[len('SSH-2.0-'):]
+        # This is an inefficient regex, but it's simple to build.
+        softwareVersionRegex = (
+            r'^(' +
+            '|'.join(
+                re.escape(c) for c in string.printable
+                if c != '-' and not c.isspace()) +
+            r')*$')
+        self.assertRegex(softwareVersion, softwareVersionRegex)
```

Comment:
This could also be:

```python
self.assertEqual(set(softwareVersion), set(string.printable) - set(string.whitespace))
```

But a failed assertion would probably be less readable.

Reply:
Right, that was my thought too.  (A failed assertion with my version is a bit wall-of-regex, but I think it's reasonably possible for somebody unfamiliar with it to work out what it means.)